File: src/medium/SortArray.swift
func sortArray(_ nums: [Int]) -> [Int] {
  guard !nums.isEmpty else {
    return []
  }
  var result = [Int]()
  result.reserveCapacity(nums.count)
  var index = 0
  var max = nums[0]
  var min = nums[0]
  var freq: [Int: Int] = [:]
  for num in nums {
    freq[num, default: 0] += 1
    if num < min { min = num }
    if num > max { max = num }
  }
  var i = min
  while i <= max {
    if let count = freq[i], count > 0 {
      result.append(contentsOf: Array(repeating: i, count: count))
    }
    i += 1
  }
  return result
}

File: lib/medium/sort_array.dart
List<int> sortArray(List<int> nums) {
  if (nums.isEmpty) return [];
  Map<int, int> freq = {};
  List<int> result = List.filled(nums.length, 0);
  int max = nums[0];
  int min = nums[0];
  int index = 0;
  for (int n in nums) {
    freq[n] = (freq[n] ?? 0) + 1;
    if (n > max) max = n;
    if (n < min) min = n;
  }
  for (int i = min; i <= max; i++) {
    int count = freq[i] ?? 0;
    while (count-- > 0) {
      result[index++] = i;
    }
  }
  return result;
}
